### PR TITLE
fix: check body size === 0 instead of null for isEmpty()

### PR DESCRIPTION
**Pull request by markdorison**
## Description
fix: check body size === 0 instead of null for isEmpty()

## Motivation / Context
Fix edge case

## Testing Instructions / How This Has Been Tested
Tests should pass! Blocked by #54

### CODE REVIEW

File: src/Http/Response.php [hunk 1]
```diff
@@ -102,7 +102,7 @@ public function getBody(): StreamInterface
      */
     public function isEmpty(): bool
     {
-        return is_null($this->response->getBody()->getSize());
+        return $this->response->getBody()->getSize() === 0;
```
Comment (apotek) [inline src/Http/Response.php]: According to the interface, `getSize()` can return `int|null`

     /**
     * Get the size of the stream if known.
     *
     * @return int|null Returns the size in bytes if known, or null if unknown.
     */
    public function getSize(): ?int;

So I think the emptiness check, should maybe be something like:

```
return $this->response->getBody()->getSize() === 0 || $this->response->getBody()->getSize() === NULL;
```

but I kind of hate that we are evaluating `$this->response->getBody()->getSize()` twice in that case. It feels a little like cheating, but would `return empty($this->response->getBody()->getSize()` be ok? I mean, I _do_ think a response of `[]` or `''` should also be considered empty even if those will never be returned from `getSize()`. 🤔

Comment (apotek) [reply src/Http/Response.php]: In any case, we should cover the possibility that it be NULL too.

If you'd like me to bring this over the line, just say the word.

**Review by apotek — APPROVED** [review]
We have to comply with the `getSize()` function signature.

**Review by apotek — CHANGES_REQUESTED** [review]
We should cover the possibility that `getSize()` be NULL too.

**Review by apotek — APPROVED** [review]
Perfect! Thanks also for adding the extra test and improving the doc block.
